packages/markdown/remark/src/rehype-collect-headers.ts

```diff
@@ -17,22 +18,18 @@ export default function createCollectHeaders() {
 				if (!level) return;
 				const depth = Number.parseInt(level);
 
-				let raw = '';
 				let text = '';
 				let isJSX = false;
-				visit(node, (child) => {
-					if (child.type === 'element') {
+				visit(node, (child, _, parent) => {
+					if (child.type === 'element' || ['code', 'pre'].some((tag) => parent?.tagName === tag)) {
```

Comment:
please use an object or Set here to avoid an unnecessary O(2) lookup.

Reply:
changed to use `Set` to scan `parent.tagName` in https://github.com/withastro/astro/pull/3502/commits/fb695313fc52428fad66ae04ea3f3bd96c8f8897